### DISCUSSION TITLE
How to add all app's in appNames?

### DISCUSSION THREAD

**Discussion by VadimBoev** [wiresock/proxifyre#41] [Ideas]
It sounds crazy, but the idea is that all applications and programs that work with the network send traffic to a proxy server, I need it. I think we need to add the '*' parameter to the appNames to enable this idea.

**Comment by wiresock** [comment]
Have you considered using an empty string instead of `*`? The matching function provided below is supposed to always return `true` when given an empty string.

```cpp
/**
 * @brief Matches an application name pattern against the process details.
 *
 * This function determines whether the application name pattern includes a path (by checking for "/" or "\\").
 * If a path is present in the pattern, the function matches it against the process's `path_name`.
 * Otherwise, it matches the pattern against the process's `name`. The comparison is case-insensitive.
 *
 * @param app The application name or pattern to compare against the process details.
 * @param process The process details to be checked against the application pattern.
 * @return true if the process details match the application pattern, false otherwise.
 */
static bool match_app_name(const std::wstring& app, const std::shared_ptr<iphelper::network_process>& process)
{
    return (app.find(L'\\') != std::wstring::npos || app.find(L'/') != std::wstring::npos)
        ? (to_upper(process->path_name).find(app) != std::wstring::npos)
        : (to_upper(process->name).find(app) != std::wstring::npos);
}
```

**Reply by VadimBoev** [reply]
I've tried these options and they all don't work:
1. "appNames": [""],
2. "appNames": [],
3. when delete "appNames", then "Topshelf.Hosts.ConsoleRunHost Error: 0 : An exception occurred, System.NullReferenceException: Ссылка на объект не указывает на экземпляр объекта."

**Reply by kirill-ant** [reply]
Actually seems like the **first one does** work - i start a sniffer on a proxy side and i see some (obviously DNS) proxified UDP requests there, but then something goes wrong here and i see that my test machine cant resolve anything. It seems like we can't just *transparrent* proxify the whole systems traffic for some reason i can't get yet.

**Comment by kirill-ant** [comment]
Same here. And no, it doesn't sounds crazy actually - windows itselves still isn't able to use socks5 with auth unfortunately. So if we need for some reason proxify all the traffic to the socks5 we need somethind like that
